crates/emulated/src/field_ops.rs

```diff
@@ -435,11 +435,11 @@ where
         if a.is_constant() && b.is_constant() {
             let a_int = BigInt::from(a);
             let b_int = BigInt::from(b);
-            let mut res_int = a_int - b_int;
-            while res_int < BigInt::zero() {
-                res_int += P::modulus();
-            }
-            res_int = res_int.rem(P::modulus());
+            let res_int = if a_int > b_int {
+                (a_int - b_int).rem(P::modulus())
+            } else {
+                P::modulus() - (b_int - a_int).rem(P::modulus())
+            };
```

Comment:
The code I had suggested had a minor bug. When `a_int == b_int`, then the value of `res_int` becomes `P::modulus()`. We want it to be zero. Please change the `>` in the `if` condition `>=`, i.e.
``` rust
let res_int = if a_int >= b_int {
```
When the value of a `BigInt` is `P::modulus()`, it does not get reduced to zero when it is converted to an `EmulatedFieldElement`. See the [comment](https://github.com/lurk-lab/bellpepper-gadgets/blob/5af3900a3114b035e3392e061a9054b4439611c9/crates/emulated/src/field_element.rs#L122) here

Sorry about missing that corner case earlier.

Reply:
You're right. I think because inside `assert_is_equal` the constants get unconditionally reduced when compared and also if it's compared to a variable, that in practice both `0` and `P::modulus()` should be indistinguishable when used from the rest of the code (i.e. I _think_ all operations would always work the same with both values)

Regardless, it's better if we get `0` here instead for the `a == b` case for sure, so I've changed it to `>=`